addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -220,29 +218,25 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
         async _showDecreaseQuantityPopup() {
             const { confirmed, payload: inputNumber } = await this.showPopup('NumberPopup', {
                 startingValue: 0,
-                title: this.env._t('Decrease the quantity by'),
+                title: this.env._t('Set the new quantity'),
             });
-            let decreaseQuantity = Math.abs(inputNumber);
-            if (confirmed && decreaseQuantity) {
+            let newQuantity = inputNumber !== ""? Math.abs(inputNumber): null;
+            if (confirmed && newQuantity !== null) {
                 let order = this.env.pos.get_order();
                 let selectedLine = this.env.pos.get_order().get_selected_orderline();
+                let currentQuantity = selectedLine.get_quantity()
 
-                let quantity = 0;
-                order.get_orderlines().forEach(orderLine => {
-                    if(orderLine.get_product().id === selectedLine.get_product().id && orderLine.get_discount() === selectedLine.get_discount())
-                        quantity += orderLine.get_quantity();
-                });
+                if(currentQuantity === 1 && newQuantity > 0)
+                    selectedLine.set_quantity(newQuantity);
+                else if(newQuantity >= currentQuantity)
```

Comment:
Can this be simplified to:
```js
if (newQuantity > currentQuantity) {
  // compute and set negative quantity to a new line
} else {
  selectedLine.set_quantity(newQuantity);
}
```

Reply:
We have to consider the case of new product, so when it's equal to 1. Otherwise, we can't allow the user to modify from 1 to 0.4 for example.

Reply:
Okay. The basic concept is that decreasing is not allow. But incrementing is okay.